### ISSUE TITLE
Token refresh does not seem to work

### ISSUE THREAD

**Issue by Bewalticus** [AxaFrance/oidc-client#1147]
### Issue and Steps to Reproduce

When our access token times out after 30 minutes it is not refreshed. Instead we get the "session timeout" error page.

This is our configuration:

```
  const configuration: OidcConfiguration = {
    client_id: oidc.clientId,
    redirect_uri: window.location.origin + '/authentication/callback',
    scope: 'openid',
    authority: oidc.authority,
    service_worker_relative_url: '/OidcServiceWorker.js',
    service_worker_only: true,
    refresh_time_before_tokens_expiration_in_second: 120,
  };
```

### Versions
7.6.0

#### Expected
The token should be renewed transparently to the application

#### Actual
The "Session Timeout" error page is shown.

### Additional Details

- Installed packages:

**Comment by guillaume-chervet** [comment]
Hi @Bewalticus ,

You need to add scope offline_access: https://github.com/AxaFrance/oidc-client/blob/4f9b3c0b767d4b955223febcc833e6ff47e997ef/packages/react-oidc/README.md?plain=1#L175

**Comment by Bewalticus** [comment]
I added the offline_access scope and the occurrence of the timeout errors is significantly reduced.
We still have them here and there but need to analyze the reason first.

**Comment by guillaume-chervet** [comment]
Thank you for your feedback @Bewalticus .
FAQ may help you https://github.com/AxaFrance/oidc-client/blob/main/FAQ.md

**Comment by VladimirPlatonenko** [comment]
We also have timeout errors time-to-time. We can't find any pattern or way to reproduce it. 
Seems like staying away from computer or letting it go to sleep mode increases chance of this happening.  Can it be any race condition? 
Any ideas how to diagnose this and trace what causes it? I tied enabling silent login but it does not help. We are using both Auth0 and Keycloak.

**Comment by guillaume-chervet** [comment]
Hi @Bewalticus , @VladimirPlatonenko , very sorry for the delay. Yes in some scenario, for example after an OS sleep mode. Internet connexion is slow to reconnect and internally oidc-client fail to refresh tokens many times.

**Comment by guillaume-chervet** [comment]
I think i can enhance the behavior by waiting an Internet connexion.

**Comment by Bewalticus** [comment]
@guillaume-chervet We see this behavior mostly on smartphones after some idle time. So I would like to go for a test with your „connection fix“.

**Comment by VladimirPlatonenko** [comment]
> Hi @Bewalticus , @VladimirPlatonenko , very sorry for the delay. Yes in some scenario, for example after an OS sleep mode. Internet connexion is slow to reconnect and internally oidc-client fail to refresh tokens many times.

Yes this seems consistent with what we see. I am keen to test connection fix when it is ready. In the mean time we are mitigating this problem via enabling silent login as last resort. This is not ideal as it relies on cookies behavior that will soon be blocked by browser,

**Comment by guillaume-chervet** [comment]
Yes, that is why i have implement Demonstrating Proof of Possession (DPoP). It will allow to increase a lot security whithout the need of Service Worker.

**Comment by guillaume-chervet** [comment]
hi @Bewalticus , @VladimirPlatonenko ,
latest version may enhance the behavior.
